servicemodules/discordConstants.py

```diff
@@ -0,0 +1,66 @@
+off_topic_channel_id: int = {"production": 997857170647433257, "development": -2}[ENVIRONMENT_TYPE]
+memes_channel_id: int = {"production": 843628797270294568, "development": -2}[ENVIRONMENT_TYPE]
+
+other_channels_category_id: int = {"production": 677546901339504642, "development": 783123903382814721}[ENVIRONMENT_TYPE]
+ai_channel_id: int = {"production": 817511894671294524, "development": 817517982832918558}[ENVIRONMENT_TYPE]
+not_ai_channel_id: int = {"production": 757908690165694546, "development": 817518001237655552}[ENVIRONMENT_TYPE]
+events_channel_id: int = {"production": 789918737415012384, "development": 817518818951430184}[ENVIRONMENT_TYPE]
+projects_channel_id: int = {"production": 787117276960522280, "development": 817518806472589342}[ENVIRONMENT_TYPE]
+book_club_channel_id: int = {"production": 929823603766222919, "development": -2}[ENVIRONMENT_TYPE]
+dialogues_with_stampy_channel_id: int = {"production": 1013976966564675644, "development": -2}[ENVIRONMENT_TYPE]
+meta_channel_id: int = {"production": 741332060031156296, "development": 817518780509847562}[ENVIRONMENT_TYPE]
+
+archive_category_id: int = {"production": 929823542818766948, "development": -2}[ENVIRONMENT_TYPE]
+voice_context_channel_id: int = {"production": 810261871029387275, "development": -2}[ENVIRONMENT_TYPE]
+editing_channel_id: int = {"production": 990947616252657667, "development": -2}[ENVIRONMENT_TYPE]
+
+test_channel_id: int = {"production": -99, "development": 803448149946662923}[ENVIRONMENT_TYPE]
+bot_owner_dms_id: int = {"production": -1, "development": 736241264856662038}[ENVIRONMENT_TYPE] #TODO: replace -1 with the id for robs DM with stampy,
+
+# TODO: it would be good if we had a test that warned us if the above did not actually match up with the server layout
+
+#################################
+# ID DERIVED VARIABLES
+#################################
+
+automatic_question_channel_id = general_channel_id  # TODO: should this be ai_safety_questions_channel_id?
+
+stampy_control_channel_ids: Tuple[int, ...] = (test_channel_id, stampy_dev_priv_channel_id, stampy_dev_channel_id,
+                                               talk_to_stampy_channel_id, bot_owner_dms_id)
+
+
+#################################
+# USER/ROLE IDs (not complete due to lack of need)
+#################################
+
+bot_admin_role_id: int = {"production": 819898114823159819, "development": 948709263461711923}[ENVIRONMENT_TYPE]
+bot_dev_role_id: int = {"production": 736247946676535438, "development": 817518998148087858}[ENVIRONMENT_TYPE]
+can_invite_role_id: int = {"production": 791424708973035540, "development": -99}[ENVIRONMENT_TYPE]
+member_role_id: int = {"production": 945033781818040391, "development": 947463614841901117}[ENVIRONMENT_TYPE]
+# pretty sure can-invite is deprecated, but putting it here for completeness
+
+rob_id: int = 181142785259208704
+stampy_id: int = "736241264856662038"
```

Comment:
please enable type checking in your editor - if you prefer explicit types, they should be at least correct ;)

I myself prefer implied types whenever possible
anyway, can we use an int value here or was there some purpose why it was a string in previous code?

Reply:
lol, i do have type checking enabled, but the way typing errors are displayed are perhaps a bit too subtle and i should mess around with the options.

the id should definitely be an int. pushed a fix for this